daemon/main.go

```diff
@@ -127,6 +132,27 @@ func main() {
 	}
 }
 
+func setupPidFile() {
+	pidfile.SetPidfilePath(defaults.PidFilePath)
+	if err := pidfile.Write(); err != nil {
+		log.WithError(err).Fatal("could not create pidfile")
+	}
+	cleanUp()
+}
+
+func cleanUp() {
+	// Handle the cleanup
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGQUIT, syscall.SIGINT, syscall.SIGHUP, syscall.SIGTERM)
+	go func() {
+		for s := range sig {
+			log.Infof("exiting received signal %v", s)
```

Comment:
You could structure this with `log.WithField("signal", s).Info("Exiting due to signal")`. Also, I think our convention is to capitalise the first letter of log lines, if you fancy being consistent with that.

Reply:
Is there some Go tooling that can detect or enforce this? Would be nice to be warned before / after compiling.

Reply:
Not easily, no. For the capitalisation thing we could get way with a regex but for the function usage the best I can think of is having a rule for `eg` and see if it changes the code as part of CI (the way we do with `go fmt`).
`eg` https://rakyll.org/eg/

Reply:
Could probably be added to golint though.